Deprecate metric cortex_distributor_sample_delay_seconds (#7516)
grafana/mimir: CHANGELOG.md

* [CHANGE] Store-gateway: Use a shorter TTL for cached items related to temporary blocks. #7407
* [CHANGE] Standardise exemplar label as "trace_id". #7475
* [CHANGE] The configuration option `-querier.max-query-into-future` has been deprecated and will be removed in Mimir 2.14. #7496

> * [CHANGE] Distributor: the metric `cortex_distributor_sample_delay_seconds` is now deprecated and will be removed in Mimir 2.14. #7516

Author: ```suggestion
* [CHANGE] Distributor: the metric `cortex_distributor_sample_delay_seconds` has been deprecated and will be removed in Mimir 2.14. #7516
```
